### PR TITLE
docs: add disallow-username-in-url.d and haproxy-protocol.d on the list

### PR DESCRIPTION
**Pull request by kdudka**
... to make make the files appear in distribution tarballs

### CODE REVIEW

**Comment by kdudka** [comment]
I believe that 26a8eeaf192c7a023ec7610bf3c13ca6248c6bd2 would do the job for us but I am not able to verify it right now because Travis CI for some reason does not see newly created branches on my fork of curl.

File: .travis.yml [hunk 1]
```diff
@@ -308,6 +308,7 @@ script:
             rm -rf curl-99.98.97
             # verify out-of-tree build
             (tar xf curl-99.98.97.tar.gz && \
+             touch docs/{cmdline-opts,libcurl}/Makefile.inc && \
```
Comment (bagder) [inline .travis.yml]: this touches in the wrong dir, it needs to be prefixed with `curl-99.98.97`. But I don't think this triggers an error if there's one or two .d files missing in the cmdline-opts directory, does it?

**Comment by kdudka** [comment]
On Friday, August 10, 2018 9:16:00 AM CEST Daniel Stenberg wrote:
> bagder commented on this pull request.
> 
> > @@ -308,6 +308,7 @@ script:
>              rm -rf curl-99.98.97
>              # verify out-of-tree build
>              (tar xf curl-99.98.97.tar.gz && \
> +             touch docs/{cmdline-opts,libcurl}/Makefile.inc && \
> 
> this touches in the wrong dir, it needs to be prefixed with `curl-99.98.97`.

Indeed, my mistake.

> But I don't think this triggers an error if there's one or two .d files
> missing in the cmdline-opts directory, does it?

It will force man pages to be regenerated and the subsequent run of `make test`
should fail on test 1139.  At least it failed on my Fedora builds when I tried
to backport the fix for #2333.

I hope I will be able to test it on Travis CI later today and confirm it...

**Comment by bagder** [comment]
Ah yes, you're right!

**Comment by kdudka** [comment]
I have figured out how to trigger Travis CI jobs when I need to (opening PR against my own fork of curl did the trick).  Unfortunately all my jobs, as well as regular upstream jobs, are now failing for unrelated reasons (some apt-get related failures).  I will give it another try after the weekend...

**Comment by bagder** [comment]
We saw those apt failures in some other PRs too today. It looks like travis-side issues.

**Comment by kdudka** [comment]
I think I have updated `.travis.yml` correctly this time.  See how the CI failure looks like:
https://travis-ci.org/kdudka/curl/jobs/415413873#L7400

**Comment by bagder** [comment]
That looks great!

**Pull request merged**